backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java

```diff
@@ -278,23 +278,28 @@ public FutureExperimentRunDAO(
 
     return checkPermission(
             Collections.singletonList(runId), ModelDBActionEnum.ModelDBServiceActions.READ)
-        .thenCompose(unused -> metricsHandler.getKeyValues(runId), executor);
+        .thenCompose(
+            unused -> metricsHandler.getKeyValues(runId, Collections.emptyList(), true), executor);
   }
 
   public InternalFuture<List<KeyValue>> getHyperparameters(GetHyperparameters request) {
     final var runId = request.getId();
 
     return checkPermission(
             Collections.singletonList(runId), ModelDBActionEnum.ModelDBServiceActions.READ)
-        .thenCompose(unused -> hyperparametersHandler.getKeyValues(runId), executor);
+        .thenCompose(
+            unused -> hyperparametersHandler.getKeyValues(runId, Collections.emptyList(), true),
+            executor);
   }
 
   public InternalFuture<List<KeyValue>> getAttributes(GetAttributes request) {
     final var runId = request.getId();
+    final var keys = request.getAttributeKeysList();
+    final var getAll = request.getGetAll();
 
     return checkPermission(
             Collections.singletonList(runId), ModelDBActionEnum.ModelDBServiceActions.READ)
-        .thenCompose(unused -> attributeHandler.getKeyValues(runId), executor);
+        .thenCompose(unused -> attributeHandler.getKeyValues(runId, keys, getAll), executor);
```

Comment:
while getting attributes via `getAttributes ` forget to considered `getAttributeKeysList`, and `getAll` flag which is added now.